services/QuillLMS/app/controllers/application_controller.rb

```diff
@@ -131,8 +131,8 @@ def non_standard_route_redirect?(route)
     flash[:error] = t('actioncontroller.errors.invalid_authenticity_token')
 
     respond_to do |format|
-      format.html { redirect_back(fallback_location: root_path) }
       format.json { render json: { redirect: URI.parse(request.referer).path }, status: 303 }
+      format.html { redirect_back(fallback_location: root_path) }
```

Comment:
Can you say a little bit more about why the reordering is necessary?  I didn't think order mattered format blocks.  Maybe related, why is reordering not needed in the method [reset_session_and_redirect](https://github.com/empirical-org/Empirical-Core/blob/a7eef4060bde74528a0729e6ebffb75d33c1e4a1/services/QuillLMS/app/controllers/application_controller.rb#L172)

Reply:
We discussed in Slack but just for posterity - turns out `fetch` sends a default `Accept` header of *%* in its requests, which tell Rails it doesn't care what format it is, so it just sends back whatever's first. I've changed it now so I undo the order reset and just add `application/json` as an explicit `Accept` header.

Reply:
Nice sleuthing on this.  I like the explicit `Accept` header.

For my future self, a couple of links from offline discussion:
https://www.bigbinary.com/blog/order-of-format-matters-in-respond_to-block
https://apidock.com/rails/ActionController/MimeResponds/InstanceMethods/respond_to#500-Ordering-of-format-blocks-is-important